### ISSUE TITLE
question: figuring out what "this" is in different contexts

### ISSUE THREAD

**Issue by tony-kerz** [iron-meteor/iron-router#348]
hey, i'm relatively new to javascript, and i'm having some trouble understanding what `this` is in various spots in the example code, i'm almost backing into it, looking at what is being called on `this` to try and figure out what it is, but even then, since there is no documentation specifying iron-router objects and methods available on them i'm still a bit lost. unless maybe `this` is a meteor object, in which case i could look at the meteor doc which has some info like that...

for instance, i was trying to figure out what `this` is in a function passed to a `before` hook and ultimately what methods i had at my disposal in that context.

so basically my question is, how am i supposed to figure out what `this` is in different contexts? is it in the doc somewhere? am i supposed to run in a browser and use a debugging tool to examine `this`? should i use console.log? look in the source code? etc...

any guidance for the weakly typed challenged appreciated ;)
